crates/nu-parser/src/parser.rs

```diff
@@ -5564,6 +5564,51 @@ pub fn parse_record(working_set: &mut StateWorkingSet, span: Span) -> Expression
             let value = parse_value(working_set, tokens[idx].span, &SyntaxShape::Any);
             idx += 1;
 
+            let bareword_error = |string_value: &Expression| {
+                let string_span = working_set.get_span_contents(string_value.span);
+                let colon_position = string_span
+                    .iter()
+                    .find_position(|b| **b == b':')
+                    .map(|(i, _)| string_value.span.start + i);
+                if let Some(colon_position) = colon_position {
+                    Some(ParseError::InvalidLiteral(
```

Comment:
I'm not sure this matches the intention of the `InvalidLiteral` error variant. Does another variant fit better or should we maybe introduce a new variant?